osu.Game/Beatmaps/ControlPoints/ControlPoint.cs

```diff
@@ -15,11 +15,9 @@ public abstract class ControlPoint : IComparable<ControlPoint>, IDeepCloneable<C
         /// The time at which the control point takes effect.
         /// </summary>
         [JsonIgnore]
-        public double Time => controlPointGroup?.Time ?? 0;
+        public double Time { get; set; }
```

Comment:
@smoogipoo during your re-review, this may be a line of note.

rationale behind this being okay is that we never update group's `Time` (and `ControlPoints` are always `AttachGroup`ed), so they can't get out of sync.